cilium/cmd/monitor.go

```diff
@@ -89,9 +94,22 @@ func setupSigHandler() {
 
 // openMonitorSock attempts to open a version specific monitor socket It
 // returns a connection, with a version, or an error.
-func openMonitorSock() (conn net.Conn, version listener.Version, err error) {
+func openMonitorSock(path string) (conn net.Conn, version listener.Version, err error) {
 	errors := make([]string, 0)
 
+	// try the user-provided socket
+	if path != "" {
+		conn, err = net.Dial("unix", path)
+		if err == nil {
+			version = listener.Version1_2
+			if strings.HasSuffix(path, "monitor.sock") {
+				version = listener.Version1_0
```

Comment:
Why is this `Version1_0` instead of `Version1_2`?

Reply:
We don't have a good way to probe protocol versions by opening a socket and attempting to use a particular version, and this seemed like the simplest approach that does the right thing.

Historically we have potentially opened one of two files:
* `/var/run/cilium/monitor.sock` - the v1.0 listener
* `/var/run/cilium/monitor1_2.sock` - the v1.2 listener

I presume that if we ever determine that we should open a new socket path for a new protocol version, we will use a new filename something like `/var/run/cilium/monitorX_Y.sock`. So, here I opted to assume we're running the latest (v1.2) protocol then only if the filename ends with the v1.0 default filename, guess that it's v1.0.

Reply:
Got it, I didn't realize that `monitor.sock` meant that it was the `v1.0` default / the versioning scheme we have now. LGTM!